src/app/services/login.service.ts

```diff
@@ -0,0 +1,32 @@
+import {HttpResponse, HttpErrorResponse} from '@angular/common/http';
+import {Injectable} from '@angular/core';
+import {Observable, of} from 'rxjs';
+import {catchError} from 'rxjs/operators';
+
+import {environment} from '../../environments/environment';
+import {User} from '../models/user';
+
+import {HttpStalkerService} from './http-stalker.service';
+@Injectable({
+  providedIn: 'root',
+})
+export class LoginService {
+  private loginURL = environment.apiUrl + '/user/login';
+
+  constructor(private httpStalker: HttpStalkerService) {}
+
+  login(user: User): Observable<HttpResponse<User>> {
+    const handler: (
+      err: HttpErrorResponse,
+      caught: Observable<HttpResponse<User>>,
+    ) => Observable<HttpResponse<User>> = this.handleError<HttpResponse<User>>();
+    return this.httpStalker.userPost(this.loginURL, user).pipe(catchError(handler));
+  }
+
+  handleError<T>(result?: T) {
+    return (error: HttpErrorResponse): Observable<T> => {
+      console.error(error);
+      return of(result as T);
+    };
+  }
+}
```

Comment:
implementare correttamente il decorator pattern. questo servizio dovrebbe solo esporre un metodo post che prende mail e password (già hashata), effettua la richiesta e restituisce un oggetto User:

```suggestion
LoginService extends StalkerHttpClientDecorator {
// ...

post(email:EmailType, hashedPassword: string): User {
// ...
}
}
```

nel caso di endpoint che hanno più di un metodo (non solo post), vanno dichiarati tutti i metodi rilevanti. in questo modo abbiamo un servizio per ogni endpoint, con 1 metodo per ogni operazione e tutti gli aspetti comuni si trovano dentro a StalkerHttpClient